src/main/java/haste/data/Storage.java

```diff
@@ -0,0 +1,136 @@
+package haste.data;
+
+import haste.commands.Parser;
+import haste.tasks.Deadline;
+import haste.tasks.Event;
+import haste.tasks.Task;
+import haste.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+/**
+ * Represents a storage for data to be stored.
+ */
+public class Storage {
+    // create data file
+    private final String filePath;
+    private File file;
+
+    /**
+     * Creates a Storage and a file for storing data if not existing.
+     *
+     * @param filePath Location of file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        create();
+    }
+
+    /**
+     * Creates a file if not existing.
+     */
+    public void create() {
+        this.file = new File(this.filePath);
+
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+                System.out.println("error creating file");
+            }
+        }
+    }
+
+    // read data file
+
+    /**
+     * Reads the data of tasks stored in file and adds the corresponding tasks into a TaskList.
+     *
+     * @param tasks TaskList where tasks are added.
+     */
+    public void read(TaskList tasks) {
+        try {
+            Scanner scanner = new Scanner(this.file);
+            if (scanner.hasNextLine()) {
+               System.out.println("loading tasks from saved files...");
+            }
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                Task currTask = readTask(line);
+                tasks.addTask(currTask);
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+    // add one line to text file
```

Comment:
Not necessarily a violation - but confusing to read. What does this line (// add one line to text file) refer to?